improving the docs and README.md (#209)
adorsys/keycloak-ssi-deployment: README.md

Reviewer: Please review this file again. I noticed there are sections based on the submodule that need to be removed or updated

Also the newly added commands should be mentioned in this file.

---

### Using the wrapper CLI

From the repository root you can use the wrapper script instead of calling the submodule directly:

```bash
./keycloak-ssi.sh setup                               # Start the OID4VCI test deployment (DB + Keycloak via submodule)
./keycloak-ssi.sh config                              # Configure realm, clients, keys, and users in the running Keycloak
./keycloak-ssi.sh test preauth IdentityCredential     # Run a pre-authorized code flow test for a given credential type
./keycloak-ssi.sh test authcode IdentityCredential    # Run an authorization code flow test for a given credential type
./keycloak-ssi.sh terraform destroy -auto-approve     # Run Terraform destroy against the configured Keycloak
./keycloak-ssi.sh terraform apply -auto-approve       # Run Terraform apply against the configured Keycloak

> ./keycloak-ssi.sh addClientScopes                     # Create and assign client scopes using the direct API helper

Reviewer: Not sure if this command alaso assigns created scopes to the client.Could you please verify?

---

### Quick setup

From the repository root, run in order:

```bash
./keycloak-ssi.sh setup    # Start DB + Keycloak (wait until ready)
./keycloak-ssi.sh config   # Configure realm, clients, keys, users
```

> Then run a credential test, e.g. `./keycloak-ssi.sh test preauth IdentityCredential`.

Reviewer: ```suggestion
Then run a credential test, for example:
```./keycloak-ssi.sh test preauth IdentityCredential.```
```

---

#### Adding custom client scopes

You can add your own client scopes without touching the submodule by:

1. Creating a `client-scopes.json` file at the **repository root**.  
   - Use the same JSON structure as the scope definitions under  
     `infrastructure/terraform/jsons/scopes/*.json` (name, protocol, protocolMappers, attributes, etc.).

>    - Each entry in the top-level JSON array represents one client scope to create..

Reviewer: ```suggestion
   - Each entry in the top-level JSON array represents one client scope to create.
```

2. Running:

   ```bash
   ./keycloak-ssi.sh addClientScopes
   ```

> This command reads `client-scopes.json`, **creates any missing client scopes**, and **assigns them as optional scopes** to the

Reviewer: ```suggestion
This command reads `client-scopes.json`, **creates the provided client scopes**, and **assigns them as optional scopes** to the
```

clients you configure (see below). By default those are `openid4vc-rest-api` and `oid4vc-demo-public` (skipping scopes or assignments that already exist).

You can create or edit `config-override.yaml` at the **repository root** and define which client IDs receive the optional scopes. The script reads the variable from that file and uses it when you run `./keycloak-ssi.sh addClientScopes`. Example:

```yaml
# config-override.yaml (repo root, gitignored)
add_client_scopes:
  target_clients: "openid4vc-rest-api oid4vc-demo-public"
```

> Use a space- or comma-separated list of client IDs. The wrapper syncs this file into the submodule before running, so changes take effect on the next `addClientScopes` run.

Reviewer: ```suggestion
Use a space or comma-separated list of client IDs. The wrapper syncs this file into the submodule before running, so changes take effect on the next `addClientScopes` run.
```

cd keycloak-oauth-sig/oid4vci-deployment
```

If you have any custom provider JARs in this repository’s `providers/` directory (for example the status list plugin or
the `keycloak-oid4vp-plugin-*.jar`) and want them available when using the submodule CLI, copy them into the submodule

> deployment before running scripts:

Reviewer: This should no longer be part of the documentation, as we can now add custom providers directly via our wrapper
